src/main/java/com/keboola/tableexporter/MetaFetcher.java

```diff
@@ -0,0 +1,200 @@
+                    }
+                    columnData.put("length", length);
+                    columnData.put("ordinalPosition", resultSet.getInt("COLUMN_ID"));
+                    columnData.put("primaryKey", false);
+                    columnData.put("uniqueKey", false);
+                    curColumns.put(curColumnIndex, columnData);
+                    curObject.put("columns", curColumns);
+                }
+                LinkedHashMap currentColumn = (LinkedHashMap) curColumns.get(curColumnIndex);
+                if (resultSet.getString("CONSTRAINT_TYPE") != null) {
+                    switch (resultSet.getString("CONSTRAINT_TYPE")) {
+                        case "R":
+                            currentColumn.put("foreignKeyName", resultSet.getString("CONSTRAINT_NAME"));
+                            currentColumn.put("foreignKeyTable", resultSet.getString("R_OWNER"));
+                            currentColumn.put("foreignKeyRef", resultSet.getString("R_CONSTRAINT_NAME"));
+                            break;
+                        case "P":
+                            currentColumn.put("primaryKey", true);
+                            currentColumn.put("primaryKeyName", resultSet.getString("CONSTRAINT_NAME"));
+                            break;
+                        case "U":
+                            currentColumn.put("uniqueKey", true);
+                            currentColumn.put("uniqueKeyName", resultSet.getString("CONSTRAINT_NAME"));
+                            break;
+                        default:
+                            break;
+                    }
+                }
+                curColumns.put(curColumnIndex, currentColumn);
+                curObject.put("columns", curColumns);
+                output.put(curTable, curObject);
+            }
+            try (FileWriter file = new FileWriter(outputFile)) {
+                JSONArray outputArray = new JSONArray();
+                Iterator it = output.entrySet().iterator();
+                while (it.hasNext()) {
+                    Map.Entry table = (Map.Entry)it.next();
+                    JSONArray columnsArray = new JSONArray();
+                    LinkedHashMap tableMap = new LinkedHashMap((LinkedHashMap) table.getValue());
+                    TreeMap<Integer, JSONObject> columns = (TreeMap<Integer, JSONObject>) tableMap.get("columns");
+                    Iterator columnsIterator = columns.entrySet().iterator();
+                    while (columnsIterator.hasNext()) {
+                        Map.Entry column = (Map.Entry)columnsIterator.next();
+                        columnsArray.add(column.getValue());
+                        columnsIterator.remove();
+                    }
+                    tableMap.put("columns", columnsArray);
+                    outputArray.add(tableMap);
+                    it.remove();
+                }
+                Gson gson = new GsonBuilder().setPrettyPrinting().create();
```

Comment:
Sure,  tbh I was only using it for more easily comparing outputs.

Reply:
it would seem best to have a very simple "functional" test going through the application class, and then a bunch other tests using directly the metafetcher class, and if that returns object it would compare the objects and not the files
i.e. it seems unnecessarily functional tests when it can be covered by unit tests

Reply:
done